lib/Dialect/TritonGPU/Transforms/AccelerateAMDMatmul.cpp

```diff
@@ -105,9 +105,13 @@ class BlockedToMFMA : public mlir::RewritePattern {
       if (elemType.isBF16()) {
         if (mfmaVersion == 1)
           kDim = 4;
-        if (mfmaVersion == 2)
+        if (mfmaVersion >= 2)
           kDim = 8;
       }
+      if (elemType.isFloat8E4M3FNUZ() || elemType.isFloat8E5M2FNUZ()) {
+        assert(mfmaVersion == 3);
+        kDim = 16;
+      }
       if (elemType.isInteger(8))
         kDim = 8;
```

Comment:
So on MI300, kDim for int8 is 16 when nonKDim is 32. I remember @scxiao has fixed this. So what is the story here?

Reply:
This PR is not merged yet: #368
Unfortunately, it depends on some parts of this PR :slightly_frowning_face:

Reply:
I think #368 is on top of this PR (#355).
@scxiao Can we merge #368 into this branch (`[binarman:mfma_fp8_support](https://github.com/binarman/triton/tree/mfma_fp8_support)`) and review again?